src/main/java/ua/com/juja/microservices/teams/service/TeamService.java

```diff
@@ -0,0 +1,79 @@
+package ua.com.juja.microservices.teams.service;
+
+import ua.com.juja.microservices.teams.dao.TeamRepository;
+import ua.com.juja.microservices.teams.entity.Team;
+import ua.com.juja.microservices.teams.entity.TeamRequest;
+import ua.com.juja.microservices.teams.exceptions.UserAlreadyInTeamException;
+import lombok.extern.slf4j.Slf4j;
+import ua.com.juja.microservices.teams.exceptions.UserInSeveralTeamsException;
+import ua.com.juja.microservices.teams.exceptions.UserNotInTeamException;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Inject;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * @author Andrii.Sidun
+ * @author Ivan Shapovalov
+ */
+@Service
+@Slf4j
+public class TeamService {
+    @Inject
+    private TeamRepository teamRepository;
+
+    public Team addTeam(TeamRequest teamRequest) {
+        log.debug("Started 'addTeam' TeamRequest: {}", teamRequest);
+        Date actualDate = Date.from(LocalDateTime.now().atZone(ZoneId.systemDefault()).toInstant());
+        List<String> usersInTeams = teamRepository.checkUsersActiveTeams(teamRequest.getMembers(), actualDate);
+        if (usersInTeams.size() > 0) {
+            log.warn("User(s) '{}' exists in a another teams", usersInTeams);
+            throw new UserAlreadyInTeamException(String.format("User(s) '%s' exists in a another teams", usersInTeams.toString()));
+        }
+        Team team = mappingRequestToTeam(teamRequest);
+        log.debug("Started 'Save team '{}'", team);
+        Team savedTeam = teamRepository.saveTeam(team);
+        log.info("Finished 'Save team' '{}'", team.getId());
+        log.debug("Finished 'Save team '{}'", team);
+        return savedTeam;
+    }
+
+    private Team mappingRequestToTeam(TeamRequest teamRequest) {
+        return new Team(teamRequest.getMembers());
+    }
+
+    public Team deactivateTeam(String uuid) {
+        log.debug("Started 'deactivateTeam' with uuid '{}'", uuid);
+        Team team = getUserActiveTeam(uuid);
+        log.debug("Finished 'usersInCurrentTeams' with uuid '{}'. Teams ", uuid, team.toString());
+        team.setDeactivateDate(Date.from(Instant.now()));
```

Comment:
In other places date has such format -  Date.from(LocalDateTime.now().atZone(ZoneId.systemDefault()).toInstant());